.travis.yml

```diff
@@ -4,23 +4,26 @@ matrix:
     - python: "2.7"
     - python: "3.5"
     - python: "pypy"
-      env: PYPY_VERSION="5.1.1" NO_COVERAGE=1 PYTHON_BUILD_CACHE_PATH="$HOME/pyenv-cache"
+      env: PYPY_VERSION="5.3" NO_COVERAGE=1 PYTHON_BUILD_CACHE_PATH="$HOME/pyenv-cache"
 cache:
-  directories:
-    - $HOME/.cache/pip
+  - pip
+  - directories:
     - $HOME/pyenv-cache
 
 before_install:
   # If necessary, set up an appropriate version of pypy.
-  - if [[ -n "$PYPY_VERSION" ]]; then source setup-pypy-travis.sh; fi
-  - if [[ -n "$PYPY_VERSION" ]]; then python --version 2>&1 | fgrep "PyPy $PYPY_VERSION"; fi
+  - |
+      if [[ -n "$PYPY_VERSION" ]]; then
+        source .travis/setup-pypy.sh
+        python --version 2>&1 | fgrep "PyPy $PYPY_VERSION"
```

Comment:
Can you maybe add a comment here explaining that we do this for `fgrep`'s exit code? The idea is that Travis sees a nonzero exit code and fails the build if `python` isn't the pypy version we're expecting it to be. (I should have commented this when I originally hacked together the pypy install thing, sorry.)